src/System.Security.Cryptography.X509Certificates/tests/TestEnvironmentConfiguration.Unix.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    internal static partial class TestEnvironmentConfiguration
+    {
+        static partial void DetermineCanModifyStores(ref bool canModify)
+        {
+            canModify = DetermineCanModifyStores();
+        }
+
+        private static bool DetermineCanModifyStores()
+        {
+            uint userId = Interop.Sys.GetEUid();
+            string certStoresFeaturePath = PersistedFiles.GetUserFeatureDirectory("cryptography", "x509stores");
+
+            Directory.CreateDirectory(certStoresFeaturePath);
+
+            // Check directory permissions:
+
+            Interop.Sys.FileStatus dirStat;
+            if (Interop.Sys.Stat(certStoresFeaturePath, out dirStat) != 0)
+            {
+                return false;
+            }
+
+            if (dirStat.Uid != userId)
+            {
+                return false;
+            }
+
+            if ((dirStat.Mode & (int)Interop.Sys.Permissions.S_IRWXU) != (int)Interop.Sys.Permissions.S_IRWXU)
+            {
+                return false;
+            }
+
+            string probeFilename = Path.Combine(certStoresFeaturePath, "probe.chmod");
```

Comment:
I realize that by construction this should only be called once, but just in case, any value in using something like `Guid.NewGuid().ToString("N") + ".chmod"` instead of `"probe.chmod"`?